document/AnnexE.mdk

```diff
@@ -0,0 +1,51 @@
+# Annex E (informative) Relationships among rules, correctness criteria, and policies { - #Annex-E }
+
+As defined in the Terms and Definitions section (&sect;[#terms-and-definitions]),
+a rule is a specific criterion for correctness verified by a static analysis tool,
+and a rule id is a unique, stable identifier that a tool associates with a rule.
+In this Annex, we limit our discussion to tools that associate a rule id with each rule;
+not all tools do so.
+
+Multiple tools might have rules which verify the same criterion for correctness
+(for example, "no null-dereference errors").
+However, those tools might associate different rule ids with those rules.
+This raises the question of how a person responsible for verifying policy compliance,
+such as a corporate compliance officer or an auditor from a government agency,
+can verify that a software product complies with a policy requirement such as
+"no null-dereference errors."
+
+The auditor can start by examining a log file.
+If the log file implements the the Policy Compliance Profile (&sect;[#policy-compliance-profile]),
+the auditor can see the list of rule ids for the rules that the tool executed.
+But how is the auditor to know what those rule ids mean?
+
+The first step is to locate the rule metadata, which can be done in various ways
+(see [Annex D](#Annex-D), Locating rule metadata).
+The `ruleDescriptor` (&sect;[#ruleDescriptor]) for this rule
+might include a `shortDescription` (&sect;[#ruleDescriptor-shortDescription])
+such as "Do not dereference a null pointer."
+If the description is clear enough and the auditor well-enough informed about
+software development, that might suffice, but it would be best not to rely on that.
+
+Ideally, there would exist a global, extensible set of criteria for correctness,
+similar to the Common Weakness Enumeration ([https://cwe.mitre.org/](https://cwe.mitre.org/)).
+Each correctness criterion would have a URI.
+
+In that case, a future version of the SARIF format could add a property `ruleUri` to the
```

Comment:
You might consider a different hypothetical property name for 'ruleUri'. ruleUri sounds like the endpoint contains information about the rule (the short/full description, ruleId, etc.) But I think it's meant to contain information about the 'weakness'. So perhaps something like 'weaknessUri' would be more appropriate?